fili-core/src/main/java/com/yahoo/bard/webservice/metadata/DataSourceMetadataService.java

```diff
@@ -58,61 +58,71 @@ public DataSourceMetadataService() {
+    public Set<SortedMap<DateTime, Map<String, SegmentInfo>>> getSegments(Set<DataSourceName> dataSourceNames) {
+        return dataSourceNames.stream()
                 .map(allSegmentsByTime::get)
                 .filter(Objects::nonNull)
                 .map(AtomicReference::get)
                 .collect(Collectors.toSet());
     }
 
     /**
-     * Get a set of intervals available for each column in the table.
+     * Get a set of intervals available for each column in the data source.
      *
-     * @param physicalTableName  The table to get the column and availability for
+     * @param dataSourceName  The data source for which to get the availability by column
      *
      * @return a map of column name to a set of available intervals
      */
-    public Map<String, SimplifiedIntervalList> getAvailableIntervalsByTable(TableName physicalTableName) {
-        if (!allSegmentsByColumn.containsKey(physicalTableName)) {
-            LOG.error(
-                    "Trying to access {} physical table datasource that is not available in metadata service",
-                    physicalTableName.asName()
-            );
-            throw new IllegalStateException(
-                    String.format(
-                            "Trying to access %s physical table datasource that is not available in metadata service",
-                            physicalTableName.asName()
-                    )
+    public Map<String, SimplifiedIntervalList> getAvailableIntervalsByDataSource(DataSourceName dataSourceName) {
+        if (!allSegmentsByColumn.containsKey(dataSourceName)) {
+            String message = String.format(
+                    "Datasource '%s' is not available in the metadata service",
+                    dataSourceName.asName()
             );
+            LOG.error(message);
+            throw new IllegalStateException(message);
         }
-        return allSegmentsByColumn.get(physicalTableName).get();
+        return allSegmentsByColumn.get(dataSourceName).get();
     }
 
     /**
-     * Update the information with respect to the segment metadata of a particular physical table.
-     * This operation should be atomic per table.
+     * Update the information with respect to the segment metadata of a particular data source.
+     * This operation should be atomic per dataSourceName.
      *
      * @param table  The physical table to which the metadata are referring to.
      * @param metadata  The updated datasource metadata.
+     *
+     * @deprecated  Pass the DataSourceName directly, rather than via the PhysicalTable
      */
+    @Deprecated
     public void update(PhysicalTable table, DataSourceMetadata metadata) {
+        table.getDataSourceNames().forEach(dataSourceName -> update(dataSourceName, metadata));
+    }
+
+    /**
+     * Update the information with respect to the segment metadata of a particular data source.
+     * This operation should be atomic per dataSourceName.
```

Comment:
Other than steps 5/6, all of the rest are happening _only_ using data from the input parameters, so those operations _are_ atomic (because there's no state touched outside this method). That said, yes, I can remove the "atomic" part and replace it with "both maps are updated" (or something to that effect)